code/presentation/mb-client/src/screens/home/namaz-timings.tsx

```diff
@@ -0,0 +1,137 @@
+/* eslint-disable max-lines-per-function */
+import { useRoute } from '@react-navigation/native';
+import React, { useEffect, useState } from 'react';
+import { StyleSheet, Text, View } from 'react-native';
+
+import { supabase } from '@/core/supabase';
+import type { RouteProp } from '@/navigation/types';
+
+const styles = StyleSheet.create({
+  container: {
+    flex: 1,
+    justifyContent: 'center',
+    alignItems: 'center',
+    backgroundColor: '#333', // Dark grey background color
+  },
+  text: {
+    color: 'green', // Green font color
+    marginBottom: 10,
+    fontSize: 18, // Increased font size
+    fontWeight: 'bold', // Bold text
+  },
+  timingsContainer: {
+    backgroundColor: '#FFF', // White background color for the container
+    padding: 10, // Padding around the timings
+    borderRadius: 5, // Rounded corners
+  },
+  timingsText: {
+    marginBottom: 5, // Spacing between timings
+  },
+});
+
+const NamazTimingsScreen = () => {
+  const [namazTimings, setNamazTimings] = useState<any>(null);
+  const [isLoading, setIsLoading] = useState(true);
+  const [error, setError] = useState<string | null>(null);
+
+  const { params } = useRoute<RouteProp<'NamazTimingsScreen'>>();
+  const selectedMasjidId = params.selectedMasjidId;
+
+  useEffect(() => {
+    const fetchNamazTimings = async () => {
+      try {
+        setIsLoading(true);
+        setError(null);
+
+        // Query the namaz timings data from Supabase based on the selectedMasjidId
+        const { data: namazTimingsData, error: timingsError } = await supabase
+          .from('namaz_timing')
+          .select('*')
+          .eq('masjid_id', selectedMasjidId);
+
+        if (timingsError) {
+          throw new Error('Error fetching namaz timings');
+        }
+
+        // Assuming namazTimingsData contains only one entry for the selected masjid
+
+        // Set the fetched namaz timings
+        setNamazTimings(namazTimingsData);
+        setIsLoading(false);
+      } catch (error) {
+        setIsLoading(false);
+        setError('Error');
+      }
+    };
+
+    fetchNamazTimings();
+  }, [selectedMasjidId]); // Run this effect whenever selectedMasjidId changes
```

Comment:
please dont use useEffect as its being depricated, you can use react query here instead like how userMasjid is done

Reply:
Removed UseEffect